File: quicksort/quicksort.py
def quicksort(xs):
    if xs == []:
        return []
    x = xs[0]
    xs = xs[1:]
    return quicksort(filter(lambda n: n <= x, xs)) + [x] + quicksort(filter(lambda n: n > x, xs))

print(quicksort([3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8]))

File: quicksort/quicksort.R
quicksort <- function(items) {
  if (identical(items, numeric(0))) { numeric(0) }
  else {
    init_item <- head(items, 1)
    tail_items <- tail(items, -1)
    c(quicksort(Filter(function(x) x < init_item, tail_items)), init_item, quicksort(Filter(function(x) x >= init_item, tail_items)))
  }
}
print(quicksort(c(3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8)))
